tests/processes/ProcessingTest.py

```diff
@@ -492,6 +494,50 @@ def test_yield_ignore_ranges(self):
         self.assertEqual(source_range.end.line, 1)
         self.assertEqual(source_range.end.column, 14)
 
+    def test_loaded_bears_with_error_result(self):
+        class BearWithMissingPrerequisites(Bear):
+
+            def __init__(self, section, queue, timeout=0.1):
+                Bear.__init__(self, section, queue, timeout)
+
+            def run(self):
+                return []
+
+            @classmethod
+            def check_prerequisites(cls):
+                return False
+
+        multiprocessing.Queue()
+        tmp_local_bears = copy.copy(self.local_bears['cli'])
+        tmp_local_bears.append(BearWithMissingPrerequisites)
+        cache = FileCache(self.log_printer,
+                          'coala_test_on_error',
+                          flush_cache=True)
+        results = execute_section(self.sections['cli'],
+                                  [],
+                                  tmp_local_bears,
+                                  lambda *args: self.result_queue.put(args[2]),
+                                  cache,
+                                  self.log_printer,
+                                  console_printer=self.console_printer)
+        cache.write()
+        self.assertLess(len(cache.data),
```

Comment:
Before, when I was removing files from the cache, there was a non-existent file(for other tests) that was not removed from cache because it did not exist. Now there is no worry about that, thanks for seeing this, I really forgot!